scipy/signal/spectral.py

```diff
@@ -511,11 +511,406 @@ def spectrogram(x, fs=1.0, window=('tukey',.25), nperseg=256, noverlap=None,
+    equally weighted, thereby avoiding aliasing and allowing full
+    reconstruction. Whether a choice of `window`, `nperseg`, and `noverlap`
+    satisfy this constraint can be tested with `check_COLA`.
+
+    .. versionadded:: 0.18.0
+
+    References
+    ----------
+    .. [1] Oppenheim, Alan V., Ronald W. Schafer, John R. Buck
+           "Discrete-Time Signal Processing", Prentice Hall, 1999.
+    .. [2] Daniel W. Griffin, Jae S. Limdt "Signal Estimation from Modified
+           Short Fourier Transform", IEEE 1984, 10.1109/TASSP.1984.1164317
+
+    Examples
+    --------
+    >>> from scipy import signal
+    >>> import matplotlib.pyplot as plt
+
+    Generate a test signal, a 2 Vrms sine wave corrupted by 0.001 V**2/Hz of
+    white noise sampled at 1024 Hz.
+
+    >>> fs = 1024
+    >>> N = 10*fs
+    >>> amp = 2 * np.sqrt(2)
+    >>> noise_power = 0.001 * fs / 2
+    >>> time = np.arange(N) / float(fs)
+    >>> carrier = amp * np.sin(2*np.pi*2e3*time)
+    >>> noise = np.random.normal(scale=np.sqrt(noise_power), size=time.shape)
+    >>> x = carrier + noise
+
+    Compute the STFT, and zero the components that are 10% or less of the
+    carrier magnitude.
+
+    >>> f, t, Zxx = signal.stft(x, fs, centered=True)
+    >>> Zxx = np.where(np.abs(Zxx) >= amp/10, Zxx, 0)
+
+    Convert back to a time series via inverse STFT
+
+    >>> _, xrec = signal.istft(Zxx, fs, centered=True)
+
+    Compare the cleaned signal with the original and true carrier signals.
+
+    >>> plt.plot(time, x, time, xrec.real, time, carrier)
+    >>> plt.xlim([0, 0.1])
+    >>> plt.xlabel('Time [sec]')
+    >>> plt.ylabel('Signal')
+    >>> plt.legend(['Carrier + Noise', 'Filtered via STFT', 'True Carrier'])
+    >>> plt.show()
+
+    """
+
+    noverlap = int(noverlap)
+    if noverlap < 0:
+        raise ValueError('noverlap must be a non-negative integer')
+
+    freq_axis = int(freq_axis)
+    time_axis = int(time_axis)
+    Zxx = np.asarray(Zxx)
+    if not np.iscomplexobj(Zxx):
+        Zxx = Zxx.astype(np.result_type(Zxx, 1j))
+
+    # Turn axes specified as negative index to positive
+    if freq_axis < 0:
+        freq_axis = Zxx.ndim + freq_axis
+    if time_axis < 0:
+        time = Zxx.ndim + time_axis
+    if freq_axis == time_axis:
+        raise ValueError('Must specify differing time and frequency axes!')
+
+    if Zxx.ndim < 2:
+        raise ValueError('Input stft must be at least 2d!')
+    else:
+        if time_axis != Zxx.ndim-1 or freq_axis != Zxx.ndim-2:
+            zouter = list(range(Zxx.ndim))
+            for ax in sorted([time_axis, freq_axis], reverse=True):
+                zouter.pop(ax)
+            Zxx = np.transpose(Zxx, zouter+[freq_axis, time_axis])
+
+    nseg = Zxx.shape[-1]
+    nperseg = Zxx.shape[-2]
+
+    if strict and not check_COLA(window, nperseg, noverlap):
+        raise ValueError('Window and stepsize do not satisfy the COLA '
+                         'constraint.')
```

Comment:
If you get the parameters wrong (which I've found very easy), the error mentions stepsize, which is a paramter you can't tune.